### PR TITLE
remove `purge` parameter to accounts

### PR DESCRIPTION
**Pull request by rob-solana**
#### Problem
 Bank and Accounts were both keeping track of is_squashed()

 #### Summary of Changes
 remove Bank's tracking of Accounts.is_squashed()

### CODE REVIEW

File: runtime/src/accounts.rs [hunk 1]
```diff
@@ -649,6 +646,16 @@ impl AccountsDB {
         parents
     }
 
+    fn is_squashed(&self, fork: Fork) -> bool {
+        self.fork_info
```
Comment (sakridge) [inline runtime/src/accounts.rs]: This is kind of more expensive than the old way, but maybe it doesn't matter.

Comment (rob-solana) [reply runtime/src/accounts.rs]: what's the old way?

Comment (sakridge) [reply runtime/src/accounts.rs]: checking the parent pointer on the bank.

Comment (rob-solana) [reply runtime/src/accounts.rs]: the duplication of state/logic was bugging me ;)

**Pull request merged**